tests/managers/vs/vs_tests.py

```diff
@@ -1132,3 +1132,61 @@ def test_get_local_disks_swap(self):
                 }
             }
         ], result)
+
+    def test_get_hardware_guests(self):
+        mock = self.set_mock('SoftLayer_Account', 'getHardware')
+        mock.return_value = [{
+            "accountId": 11111,
+            "domain": "vmware.chechu.com",
+            "hostname": "host14",
+            "id": 22222,
+            "virtualHost": {
+                "accountId": 11111,
+                "id": 33333,
+                "name": "host14.vmware.chechu.com",
+                "guests": [
+                    {
+                        "accountId": 11111,
+                        "hostname": "NSX-T Manager",
+                        "id": 44444,
+                        "maxCpu": 16,
+                        "maxCpuUnits": "CORE",
+                        "maxMemory": 49152,
+                        "powerState": {
+                            "keyName": "RUNNING",
+                            "name": "Running"
+                        },
+                        "status": {
+                            "keyName": "ACTIVE",
+                            "name": "Active"
+                        }
+                    }]}}]
+
+        result = self.vs.get_hardware_guests()
+
+        self.assertEqual([{
```

Comment:
Use a more precise assertion here. Just test a few fields that you expect to exist, don't need to test the entire object is strictly equal.

Reply:
Done.